x/hub-genesis/keeper/keeper.go

```diff
@@ -56,21 +56,16 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 
 // lock coins by sending them to an escrow address
 func (k Keeper) lockRollappGenesisTokens(ctx sdk.Context, sourceChannel string) error {
-	account := k.accountKeeper.GetModuleAccount(ctx, types.ModuleName)
-
 	// get spendable coins in the module account
+	account := k.accountKeeper.GetModuleAccount(ctx, types.ModuleName)
 	spendable := k.bankKeeper.SpendableCoins(ctx, account.GetAddress())
 
-	// send coins to the escrow address
-	if spendable.Empty() {
-		return types.ErrGenesisNoCoinsOnModuleAcc
+	// validate it's enough for the required tokens
+	locked := k.GetLocked(ctx)
+	if !spendable.IsAllGTE(locked.Tokens) {
```

Comment:
I'm guessing the purpose of `locked.Tokens` is to have control over how many tokens we want to lock at genesis event, instead of just locking all tokens available in the module. Is the only place where we can set the expected tokens in `NewGenesisState`? And if `DefaultGenesisState` is used, these tokens will be empty. We should perhaps have this check before proceeding to lock them. I suppose it's also valid that someone doesn't want to lock any tokens at all, but then they might not need to trigger genesis event at all.

Reply:
I;ve added the tokens so it will be easy to query and validate the genesis lock status.
As we have trust assumptions for this feature, we need to provide some way to double check it.

I've also added validation on genesis, that the genesis balance is **the same** as the expected locked tokens